tests/test_ssl_transport_close.py

```diff
@@ -0,0 +1,83 @@
+from aiohttp import web
+import aiohttp
+import asyncio
+import warnings
+
+
+async def handle_root(request):
+    """
+    respond after server was restarted, forcing connection_lost
+    """
+    cq = request.app['cq']
+    dq = request.app['cq']
+    i = await cq.get()
+    cq.task_done()
+    await dq.put(i)
+    return web.Response(text='')
+
+
+async def _prepare(aiohttp_server, ssl_ctx, client_ssl_ctx, cq, dq):
+    app = web.Application()
+    app['cq'] = cq
+    app['dq'] = dq
+    app.router.add_get('/', handle_root)
+    server = await aiohttp_server(app, port=12345, ssl=ssl_ctx)
+    asyncio.ensure_future(_restart(server.runner, ssl_ctx, cq))
```

Comment:
I don't get why you need to schedule a task on the loop and not wait for it. You're basically creating a synthetic race condition that will likely make the testing indeterministic.